pkg/reconciler/v1alpha1/clusteringress/clusteringress.go

```diff
@@ -325,6 +343,52 @@ func (c *Reconciler) reconcileVirtualService(ctx context.Context, ci *v1alpha1.C
 	return nil
 }
 
+func (c *Reconciler) ensureFinalizer(ci *v1alpha1.ClusterIngress) error {
+	finalizers := sets.NewString(ci.Finalizers...)
+	if finalizers.Has(clusterIngressFinalizer) {
+		return nil
+	}
+	finalizers.Insert(clusterIngressFinalizer)
+
+	mergePatch := map[string]interface{}{
+		"metadata": map[string]interface{}{
+			"finalizers":      finalizers.List(),
```

Comment:
does the order of the finalizers matter?

Reply:
No, the order does not matter.
An object is deleted when all of the finalizers are removed. The deletion order of finalizers does not matter.

Reply:
Then if it doesn't matter why do we check that our finalizer is the "first one" below?